src/Vector.cpp

```diff
@@ -0,0 +1,207 @@
+#include "Vector.h"
+
+
+// Constructor
+Vector::Vector(size_t size, int value)
+    : m_size(size)
+    , m_capacity(get_capacity(size))
+    , m_data(new int[m_capacity])
+{
+    // Initialize all elements to the given value
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = value;
+    }
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(new int[m_capacity])
+{
+    // Copy all elements from the other vector
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = other.m_data[i];
+    }
+}
+
+// Move constructor
+Vector::Vector(Vector&& other) noexcept
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(other.m_data)
+{
+    // Take ownership of the other vector's data
+    other.m_size = 0;
+    other.m_capacity = 0;
+    other.m_data = nullptr;
+}
+
+// Copy assignment operator
+Vector& Vector::operator=(const Vector& other) {
+    if (this != &other) {
+        // Copy and swap idiom for exception safety
+        Vector temp(other);
+        std::swap(m_size, temp.m_size);
```

Comment:
I saw that below in the code you implemented a `Vector::swap` method. Why not use it and spare the code duplication?